core/src/oauth/connection.rs

```diff
@@ -3,9 +3,29 @@ use crate::utils;
 use crate::utils::ParseError;
 use anyhow::Result;
 use async_trait::async_trait;
+use base64::{engine::general_purpose, Engine as _};
+use lazy_static::lazy_static;
+use ring::aead::NONCE_LEN;
+use ring::{
+    aead,
+    rand::{self, SecureRandom},
+};
 use serde::{Deserialize, Serialize};
-use std::fmt;
 use std::str::FromStr;
+use std::{env, fmt};
+
+lazy_static! {
+    static ref ENCRYPTION_KEY: Result<aead::LessSafeKey, &'static str> = {
+        let encoded_key =
+            env::var("OAUTH_ENCRYPTION_KEY").map_err(|_| "OAUTH_ENCRYPTION_KEY: not set")?;
```

Comment:
So the encryption key is set in the environment as a base64?

Reply:
yes as per the bin to generate it

Reply:
Yep that's what I figured but given that it was not explicit in the PR description or the spec, just wanted to confirm 🙃.